crd/overlayextensionconfig/api/v1alpha1/overlayextensionconfig_types.go

```diff
@@ -34,11 +34,14 @@ type OverlayExtensionConfigList struct {
 // +kubebuilder:validation:XValidation:rule="!has(oldSelf.extensionIPRange) || has(self.extensionIPRange)", message="ExtensionIPRange is required once set"
 type OverlayExtensionConfigSpec struct {
 	// ExtensionIPRange field defines a CIDR that should be able to reach routing domain ip addresses.
-	// +kubebuilder:validation:Optional
 	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="Value is immutable"
 	// +kubebuilder:validation:MaxLength=43
 	// 43 is max length of IPv6 CIDR string
 	ExtensionIPRange string `json:"extensionIPRange,omitempty"`
+
+	// AppGatewaySubnetID field defines a subnet delegated to Application Gateway that should be able to reach routing domain ip addresses.
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="Value is immutable"
+	AppGatewaySubnetID string `json:"appGatewaySubnetID,omitempty"`
```

Comment:
is this an ARM ID or a GUID?

Reply:
let's call it AppGatewaySubnetGUID then, just ID implies ARM ID

Reply:
nit: can we make it clear in the comment as well